daml-script/daml/Daml/Script.daml

```diff
@@ -121,6 +125,24 @@ data CommandsF a
 newtype Commands a = Commands (Ap CommandsF a)
   deriving (Functor, Applicative)
 
+data DisclosedContract = DisclosedContract
+  with
+    contractId : AnyContractId
+    argument : AnyTemplate
+    metadata : ContractMetadata
+
+data ContractMetadata = ContractMetadata
```

Comment:
We also need to expose this on `query` but happy to do that in a separate PR if that makes it easier.

Reply:
I implemented the `query` side as well, but I'd like to open a separate PR for those together with some tests.